src/squlearn/util/pennylane/pennylane_circuit.py

```diff
@@ -143,7 +143,9 @@ def __init__(
             self._executor = Executor("pennylane")
 
         # Transpile circuit to supported basis gates and expand blocks automatically
-        self._qiskit_circuit = transpile(circuit, basis_gates=qiskit_pennylane_gate_dict.keys())
```

Comment:
I can't remember, is there now no decomposition required anymore?

Reply:
I put the decomposition function at the point where a qiskit quantum circuit is build from the squelearn circuit. So in this case in lowlevel_qnn_pennylane. Then in pennylane_circuit, it is already decomposed.

Reply:
Would it be more save to put it in the pennylane_circuit.py, just in case if someone uses pennylane with blocks

Reply:
I am not familiar enough with the code, to say where the best positions to decompose the circuits are, sothat the circuit is decomposed in every function that is available in squlearn. But sofar the decompose function only works on qiskit circuits, so i put it where qiskit circuits are build. If you give me all positions where it should be to ensure the upper while being compatible with the lower, i can change the positions accordingly.